app/tests/unit/lib/theme/widgets/ChangeReasonWidget.test.tsx

```diff
@@ -0,0 +1,127 @@
+import { act, fireEvent, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import compiledChangeReasonWidgetQuery, {
+  ChangeReasonWidgetQuery,
+} from "__generated__/ChangeReasonWidgetQuery.graphql";
+import ChangeReasonWidget from "components/ProjectRevision/ChangeReasonWidget";
+import { graphql } from "react-relay";
+import ComponentTestingHelper from "tests/helpers/componentTestingHelper";
+
+const testQuery = graphql`
+  query ChangeReasonWidgetQuery($projectRevision: ID!) @relay_test_operation {
+    query {
+      # eslint-disable-next-line relay/unused-fields
+      projectRevision(id: $projectRevision) {
+        ...ChangeReasonWidget_projectRevision
+      }
+    }
+  }
+`;
+
+const mockQueryPayload = {
+  Query() {
+    return {
+      projectRevision: {
+        id: "test-revision-id",
+        changeStatus: "pending",
+      },
+    };
+  },
+};
+
+const handleChangeReason = jest.fn();
+
+const componentTestingHelper =
+  new ComponentTestingHelper<ChangeReasonWidgetQuery>({
+    component: ChangeReasonWidget,
+    testQuery: testQuery,
+    compiledQuery: compiledChangeReasonWidgetQuery,
+    getPropsFromTestQuery: (data) => ({
+      formContext: data.query,
+    }),
+    defaultQueryResolver: mockQueryPayload,
+    defaultQueryVariables: { projectRevision: "Test Revision ID" },
+    defaultComponentProps: {
+      id: "test-id",
+      value: "A reason for change",
+      options: {
+        text: "just for testing", //This is a required prop but not required for the test
+      },
+      onChange: handleChangeReason,
+    },
+  });
+
+describe("The ChangeReasonWidget", () => {
+  beforeEach(() => {
+    componentTestingHelper.reinit();
+  });
+
+  it("renders the change reason textbox widget with default value along with an action button", () => {
+    componentTestingHelper.loadQuery();
+    componentTestingHelper.renderComponent();
+
```

Comment:
Should there be an assertion for the default value? Or that doesn't need to be in the test title?